File: Hello_World_Tutorials/testing/java/bst_eq.java
import bridges.connect.Bridges;
import bridges.base.BSTElement;
import bridges.data_src_dependent.EarthquakeUSGS;
import bridges.data_src_dependent.Tweet;
import bridges.data_src_dependent.USGSaccount;
import java.util.List;

public class bst_eq {
	public static void main(String[] args) throws Exception{
    	
   					//create the Bridges object
		Bridges bridges = new Bridges(9, "YOUR_API_KEY", "YOUR_USER_ID");

		bridges.setTitle("A Binary Search Tree Example with Earthquake Data");

					// Retrieve a list of 10 earthquake  records  from USGS using the BRIDGES API
		USGSaccount name = new USGSaccount( "earthquake" );
		List<EarthquakeUSGS> eqlist = Bridges.getEarthquakeUSGSData(name, 10);

					// create BST nodes and insert into a tree
		BSTElement<Double, EarthquakeUSGS> root = null; 
		for (int k = 0; k < 10; k++) {
			BSTElement<Double, EarthquakeUSGS> bst_node = 
				new BSTElement<Double, EarthquakeUSGS>(eqlist.get(k).getMagnitude(), eqlist.get(k)); 
					// set label of the node
			bst_node.setLabel(eqlist.get(k).getTitle() + "\n\nLat/Long: ( " + 
				eqlist.get(k).getLatit() + "," + eqlist.get(k).getLongit() + " )\n\n" + 
				eqlist.get(k).getTime());
			root = insert_R (root, bst_node);
		}


								// set some visual attributes
		root.getVisualizer().setColor("red");

             					//set visualizer type
		bridges.setDataStructure(root);
        						// visualize the tree
		bridges.visualize();
    }
				// recursive insert method to insert nodes into a binary search tree
	public static BSTElement<Double, EarthquakeUSGS> insert_R (BSTElement<Double, EarthquakeUSGS> rt, 
					BSTElement<Double, EarthquakeUSGS> new_el) {
		if (rt == null)
			return new_el;
		else if (new_el.getKey() < rt.getKey())
			 rt.setLeft(insert_R(rt.getLeft(), new_el));
		else 
			 rt.setRight(insert_R(rt.getRight(), new_el));
		return rt;
	}
}

File: Hello_World_Tutorials/testing/c++/bst_eq.cpp

#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"
#include "BSTElement.h"
#include "BST.h"


using namespace std;
using namespace bridges;

int max_quakes = 25;

int main(int argc, char **argv) {
    Bridges *bridges =  new Bridges(50, "YOUR_USER_ID", "YOUR_API_KEY");

	string hilite_color = "orange", 
			def_color = "green",
			end_color = "red";
		
								// read the earth quake  data and build the BST
	bridges->setTitle("Recent Earthquakes (USGIS Data)");

    vector<EarthquakeUSGS> eq_list = DataSource::getEarthquakeData(max_quakes);

	BST<float, EarthquakeUSGS> *bst = new BST<float, EarthquakeUSGS> ();

	for (int k = 0; k < max_quakes; k++) {
		EarthquakeUSGS eq = eq_list[k];
		bst->insert(eq.getMagnitude(), eq);
	}

					// visualize the binary search tree
	bridges->setDataStructure(bst->getRoot());
	bridges->visualize();

	return 0;
}
